commcare_connect/opportunity/models.py

```diff
@@ -684,6 +685,40 @@ def __setattr__(self, name, value):
                 self.status_modified_date = now()
         super().__setattr__(name, value)
 
+    def save(self, **kwargs):
+        # The code below automatically switches/create CompletedWork when the UNIQUE_USER_VISIT_CONSTRAINT
+        # is triggered.
+        rank = kwargs.pop("rank", 0)
+        try:
+            with transaction.atomic():
```

Comment:
why wrap a transaction here?

Reply:
Without this transaction, any other transaction will exit due to the IntegrityError exception, this basically helps as a wrapper to localize that error to only this transaction instead of any other parent transactions.